plugins/restful/RestfulDataProviderVariable.php

```diff
@@ -0,0 +1,306 @@
+    return array('restful_count' => count($this->getVariablesForList()));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function index() {
+    $variables = $this->getVariablesForList();
+    $return = array();
+    foreach ($variables as $variable => $value) {
+      $return[] = $this->view($variable);
+    }
+
+    return $return;
+  }
+
+  /**
+   * Helper function to feed an array into viewMultiple().
+   *
+   * @param string $name_string
+   *  A string of variable names, separated by commas.
+   */
+  public function viewVariables($name_string) {
+    $names = array_unique(array_filter(explode(',', $name_string)));
+    return $this->viewMultiple($names);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewMultiple(array $names) {
+    if (empty($names)) {
+      return array();
+    }
+    $variables = array_keys($this->getVariablesForList());
+    $output = array();
+
+    // Build output according to filtered/sorted list of variables.
+    foreach ($variables as $variable) {
+      if (in_array($variable, $names)) {
+        $output[] = $this->view($variable);
+      }
+    }
+
+    return $output;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function view($name) {
+    $cache_id = array(
+      'tb' => 'variable',
+      'id' => $name,
+    );
+    $cached_data = $this->getRenderedCache($cache_id);
+    if (!empty($cached_data->data)) {
+      return $cached_data->data;
+    }
+
+    $return = array();
+    foreach ($this->getPublicFields() as $public_field_name => $info) {
+      if ($info['property'] == 'name') {
+        $public_field_value = $name;
+      }
+      elseif ($info['property'] == 'value') {
+        $public_field_value = variable_get($name);
+      }
+      else {
+        continue;
+      }
+
+      // Modify the public field value using a callback, if supplied.
+      $public_field_value = $info['callback'] ? static::executeCallback($info['callback'], array($value)) : $public_field_value;
+
+      // Modify the public field value using a process callback, if supplied.
+      if ($public_field_value && $info['process_callbacks']) {
+        foreach ($info['process_callbacks'] as $process_callback) {
+          $public_field_value = static::executeCallback($process_callback, array($public_field_value));
+        }
+      }
+
+      $return[$public_field_name] = $public_field_value;
+    }
+
+    $this->setRenderedCache($return, $cache_id);
+
+    return $return;
+  }
+
+  /**
+   * Alias for $this->setVariable().
+   *
+   * @param string $name
+   *  The name of the variable to set a value for.
+   */
+  public function update($name, $full_update = TRUE) {
+    $this->setVariable($name);
+  }
+
+  /**
+   * Sets a variable value.
+   *
+   * This method is used for both 'create' and 'update' contexts.
+   *
+   * @param string $name
+   *  The name of the variable to set a value for.
+   *
+   * @todo: get $name from the request: make it optional for create()/update().
```

Comment:
`$name` is ignored in the payload for the update (because it's the ID of the resource). Although it should be mandatory on create.

Please remove the `@todo` comment.